### ISSUE TITLE
Test mIoU ended up in ~0.3 after 200 epochs training

### ISSUE THREAD

**Issue by KlZOnmyway** [OptMN-Lab/sdmgrad#1]
<img width="324" alt="Screenshot 2024-10-16 122513" src="https://github.com/user-attachments/assets/fa3dc3d0-f2c3-464e-92f7-d0e9ed7dedcb">

Hi, great work! I was trying to regenerate the results of cityscape tasks. I tried both the full resolution datasets and the smaller one, however, the converged model can only obtain 0.3 in mIoU in test set. My training setting is: 
```
  lr=1e-4,
  n_epochs=2000,
  batch_size=8,
  method = 'sdmgrad',
  gpu=0,
  lamda=0.3,
  seed=0,
  niter=20,
```
Could you please tell me what's happening?

**Comment by bhsimon0810** [comment]
Hi, thanks for your interest in our work. I just tested our code with the default configuration, and the test mIoU reached 0.43 after only 1 epoch, which is better than the results you reproduced after training for 200 epochs.

![image](https://github.com/user-attachments/assets/e4870914-d7c8-4755-aa57-66f63d9f0a97)

Which model are you using, `segnet` or `mtan`? The default choice should be `mtan`. Additionally, have you made any modifications to the codes?

**Comment by KlZOnmyway** [comment]
Really thanks for the quick reply. I checked the code and found out I reduced the training loop for debugging and forgot to turn it back.
```
[1  372/372] semantic loss: 0.628, depth loss: 0.030, :   0%|          | 0/200 [23:55<?, ?it/s]LOSS FORMAT: SEMANTIC_LOSS MEAN_IOU PIX_ACC | DEPTH_LOSS ABS_ERR REL_ERR 
Epoch: 0000 | TRAIN: 1.0663 0.2319 0.6115 | 0.0601 0.0601 170.7193 | 0.7416 TEST: 0.3704 0.7286 0.0387 | 0.0387 154.8246| 184.692
```
Just one more question, for the results in the paper, are those trained on smaller cityscapes or the full size one? The above result is from smaller cityscapes dataset.

**Comment by bhsimon0810** [comment]
Hi, I just noticed that there is an update in the [MTAN](https://github.com/lorenmt/mtan) repo. For the Cityscapes dataset, our SDMGrad and other methods like [FAMO](https://github.com/Cranial-XIX/FAMO), [FairGrad](https://github.com/OptMN-Lab/fairgrad), etc., all use the smaller version. You can also find some related information in [FAMO](https://github.com/Cranial-XIX/FAMO).

**Comment by KlZOnmyway** [comment]
Thanks!
